src/main/java/laura/TaskList.java

```diff
@@ -0,0 +1,192 @@
+package laura;
+
+import laura.exception.DecodeException;
+import laura.exception.LauraException;
+import laura.task.DeadlineTask;
+import laura.task.EventTask;
+import laura.task.Task;
+import laura.task.ToDoTask;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+/**
+ * Holds a list of Tasks that can be manipulated with various methods
+ */
+public class TaskList {
+    /** Where the local data should be stored */
+    private final String dataPath;
+    /** The list that will hold the tasks*/
+    ArrayList<Task> list;
```

Comment:
Not sure I like the naming here, perhaps use tasks for a TaskList instead?
```suggestion
    /** The list that will hold the tasks*/
    ArrayList<Task> tasks;
```